src/libprojectM/Renderer/Renderer.hpp

```diff
@@ -218,7 +217,7 @@ class Renderer
     int m_viewportWidth{0};
     int m_viewportHeight{0};
 
-    std::string m_presetPath;
+    const std::vector<std::string>& m_textureSearchPaths; ///!< List of paths to search for texture files
```

Comment:
I'd avoid making member variables const, as that disables [assignment and move semantics](https://lesleylai.info/en/const-and-reference-member-variables/).

Reply:
I don't see why it shouldn't ever be moved? I'd rather have everything be moveable if there is no fundamental reason that they shouldn't be.

Reply:
This list is passed through many hands, and there's not even a proper way to update it without recreating projectM. It's not the only parameter that has this issue, so I'd rather properly rework this in a later commit. Even if it would be copied every time now, there's no real performance implication as it's basically a one-time operation (when projectM is initialized). Even writing this text here will probably take _way_ longer than properly moving this list will globally save in CPU time for all users combined.

Reply:
Converted it to a simple reference, so the list is now only stored in the `ProjectM` class. As long as the user doesn't manipulate the list while accessing it in a different thread, this should be fine. There aren't any function for that either now.